lib/Manager.php

```diff
@@ -171,4 +171,20 @@ public function createRoom($type, $name) {
 
 		return new Room($this->db, $roomId, $type, $name);
 	}
+
+	/**
+	 * @param Room $room
+	 * @return Room
+	 * @throws RoomNotFoundException
+	 */
+	public function upgradeRoom(Room $room) {
```

Comment:
Yeah i thought about this as well. The problem is there should only ever be one one2one room for two users. So in case they already have one we still will have to keep it. So i thought we should just not do more magic then necessary.
We could still change the method and simply not allow to pass a different type, but use is little...